src/java/org/apache/cassandra/metrics/CassandraMetricsRegistry.java

```diff
@@ -250,6 +271,43 @@ private void removeAlias(MetricName name)
         if (mBeanServer.isRegistered(name.getMBeanName()))
             MBeanWrapper.instance.unregisterMBean(name.getMBeanName(), MBeanWrapper.OnException.IGNORE);
     }
+
+    /**
+     * Very infrequently used meters generate a linear amount of tick work based on how long it has been
+     * since the meter was last marked or read. On scales of a year this can be enough to cause the first request
+     * that needs to mark the meter to time out. Once a day read every meter to force them to run Meter.tickIfNecessary
+     * so we only ever run at most one day worth of tick work per meter in the request path.
+     *
+     * This can be removed if we ever upgrade and switch the default MovingAverage from EWMA to SlidingWindowTimeAverages
+     */
+    private void tickMeters()
+    {
+        List<Throwable> failures = new ArrayList<>();
+        int droppedFailures = 0;
+        for (Meter meter : getMeters().values())
```

Comment:
I think it is more efficient to filter the metrics by type instead of calling the `getMeters()` method, which creates a new map each time it is called - it is expensive for a large set of metrics.

Reply:
This gets called once a day in a background thread. Assume 100 meters per table and 1000 tables it takes about 26 seconds on my laptop. This is to process a days worth of ticks on every meter as if it were unused.

So maybe `0.00003761574` of daily CPU time on an 8 core server. The time is dominated by actually ticking not iterating the metrics.

I really don't think it's worth the extra lines of code to make iteration faster.

Reply:
This is actually +2 extra lines of code! Anyway, it's not important for the workaround itself, the comment is just based on the assumption that if we can have zero impact, why shouldn't we? So I agree with you that the node can live without it and this is not a place where we should save resources :-)

Reply:
Which lines are the extra lines?  I would remove 2 extra lines if I could?

Reply:
I would change

```
for (Meter meter : getMeters().values())
{
    ...
}
```

to the following

```
for (Metric metric : getMetrics().values())
{
    if (!Meter.class.isInstance(metric))
        continue;
    ...
   
}
```